src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,959 @@
+        displayName: `Property ${index + 1} Visualization`,
+        required: false,
+        visibleDependency: `property${index}`,
+        visiblePredicate: (dependentValue: Property) =>
+            dependentValue !== Property.None,
+    }
+
+    return propertyVisualization
+}
+
+function getPropertyColor(value: string, index: number) {
+    const propertyColor = {
+        value: value,
+        forceType: 'color',
+        displayName: `Property ${index + 1} color:`,
+        required: false,
+        visibleDependency: `property${index}`,
+        visiblePredicate: (dependentValue: Property) =>
+            dependentValue !== Property.None,
+    }
+
+    return propertyColor
+}
+
+class VisualizerGrid extends VisualizerDefault {
+    name = 'Grid'
+    description =
+        'This visualizer puts any number sequence in a spiral or in '
+        + 'rows and allows you to highlight numbers based on various'
+        + ' properties.'
+        + ' One use of this tool is to put the natural numbers in a spiral'
+        + " and to highlight the prime ones, which is known as Ulam's "
+        + 'spiral. '
+        + 'Another use of this tool is highlighting properties such as'
+        + ' whether a number is abundant, polygonal, '
+        + 'or in some other sequence.'
+
+    //Grid variables
+    amountOfNumbers = 40000
+    currentSequenceIndex = 0
+    startingIndex = 0
+    currentNumber = 0n
+    showNumbers = false
+    preset = Preset.Custom
+    pathType = PathType.Spiral
+    resetAndAugmentByOne = false
+    squareValues = false
+    backgroundColor = BLACK
+    numberColor = WHITE
+
+    //Path variables
+    x = 0
+    y = 0
+    scalingFactor = 0
+    currentDirection = Direction.Right
+    numberToTurnAtForSpiral = 0
+    incrementForNumberToTurnAt = 1
+    whetherIncrementShouldIncrement = true
+
+    //Properties
+    propertyObjects = [
+        {
+            property: Property.Prime,
+            visualization: PropertyVisualization.Primary_Color,
+            color: RED,
+        },
+    ]
+
+    params: {[key: string]: ParamInterface} = {
+        preset: {
+            value: this.preset,
+            from: Preset,
+            displayName: 'Presets',
+            required: false,
+            description:
+                'Note: If a preset is selected, properties no longer function.',
+        },
+        amountOfNumbers: {
+            value: this.amountOfNumbers,
+            displayName: 'Amount of Numbers',
+            required: false,
+            description:
+                'Note: There is an increased lag time over 10,000 numbers',
+        },
+        startingIndex: {
+            value: this.startingIndex,
+            displayName: 'Starting Index',
+            required: false,
+            description: 'Note: This does not accept negative numbers.',
+        },
+        pathType: {
+            value: this.pathType,
+            from: PathType,
+            displayName: 'How path moves about grid',
```

Comment:
Kate: I suggest displayName 'Grid path' (to avoid text wrapping)

For the original comment, see https://github.com/numberscope/frontscope/pull/164#discussion_r1046400670.

Reply:
[This commit](https://github.com/numberscope/frontscope/pull/230/commits/b8fdb093b701217c19d8a13a7886748bafdcf132) should have changed this to look/read better. Someone please check and resolve this conversation if so.

Reply:
The wording is changed but on my screen it still wraps.
![Screenshot from 2022-12-29 12-44-06](https://user-images.githubusercontent.com/15117097/210004536-b919a041-d325-49b3-a610-50d436c4a460.png)

Reply:
Wow I am glad you checked. That is so weird. We are going to have to do a joint debugging session where you share screen, because it looks perfect on my end. Maybe in our next meeting if the agenda isn't otherwise too full.

Reply:
We're going to change "Shape of grid path:" and "Path in grid:" and "Rows_Reset_Augmented_Rows" to "Offset Rows".

Reply:
done in commit [cd447c6](https://github.com/numberscope/frontscope/pull/230/commits/cd447c694250537ab7694261e490f9a72832575d) ; if it looks good, please resolve.

Reply:
Check!